flow/layers/physical_model_layer.h

```diff
@@ -35,12 +39,83 @@ class PhysicalModelLayer : public ContainerLayer {
 #endif  // defined(OS_FUCHSIA)
 
  private:
-  SkRRect rrect_;
+  std::unique_ptr<PhysicalLayerShape> shape_;
   float elevation_;
   SkColor color_;
   SkScalar device_pixel_ratio_;
 };
 
+// Common interface for the shape operations needed by PhysicalModelLayer.
+class PhysicalLayerShape {
+ public:
+  virtual const SkRect& getBounds() const = 0;
+  virtual const SkPath getPath() const = 0;
+  virtual void clipCanvas(SkCanvas& canvas) const = 0;
+  virtual bool isRect() const = 0;
+#if defined(OS_FUCHSIA)
+  virtual const SkRRect& getFrameRRect() const = 0;
+#endif  // defined(OS_FUCHSIA)
+};
+
+class PhysicalLayerRRect : public PhysicalLayerShape {
+ public:
+  PhysicalLayerRRect(const SkRRect& rrect) { rrect_ = rrect; }
+
+  // |flow::PhysicalLayerShape|
+  const SkRect& getBounds() const { return rrect_.getBounds(); }
```

Comment:
You are missing the `override` declaration on these. I will try to add a `clang-tidy` check for this. Thanks for decorating the method with a comment state what this overrides though.

Reply:
done